up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,615 @@ SPDX-FileType: DOCUMENTATION
+
+[.specitem,oft-sid="req~usubscription-remote-max-timeout~1",oft-needs="impl"]
+--
+Remote Subscribe/Unsubscribe requests *MUST* implement a minimum timeout of 5 minutes.
+--
+
+[.specitem,oft-sid="req~usubscription-remote-retry-indefinitely~1",oft-needs="impl"]
+--
+Timed-out remote commands *MUST* be retried indefinitely until the business logic behind it no longer requires the command to be sent (e.g. because the last entity unsubscribed from a topic that is in state `SUBSCRIPTION_PENDING`).
+--
+
+[#reset-operation]
+== Reset
+
+This is a private API, to be used only between uSubscription services. Regular uEs can call Unsubscribe() to flush their own subscriptions.
+
+[.specitem,oft-sid="req~usubscription-reset-protobuf~1",oft-needs="impl"]
+--
+The uSubscription service `Reset()` function *MUST* implement the corresponding protobuf definition in link:../up-core-api/uprotocol/core/uSubscription/v3/usubscription.proto[usubscription.proto].
+--
+
+[.specitem,oft-sid="req~usubscription-reset~1",oft-needs="impl,utest"]
+--
+When the `Reset()` function is called, the uSubscription service *MUST* flush all stored subscription information.
+--
+
+[.specitem,oft-sid="req~usubscription-reset-only-usubscription~1",oft-needs="impl,utest"]
+--
+When receiving a `Reset()` call from a source that is not another uSubscription services (i.e. from source URIs where uEntity ID (`ue_id`) does not equal _0x8000_), uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `PERMISSION_DENIED`.
+--
 
 == uSubscription Sequences
 
 In the following section, we will elaborate on the various subscription flows for local and remote topics. When a consumer subscribes to a remote topic, it is the responsibility of the (local) uSubscription service to relay the subscription request to the remote uSubscription service as can be seen in the sequence diagrams below.
 
 NOTE: Throughout this section we will use the sample topic `//Device1/uexample/1/resource#Event` to illustrate the various sequences. The above-mentioned topic will be replaced with `_topic_` in the diagrams
 
-
-
 === Subscription
 
 Subscription flow will show how a subscriber can subscribe to the example topic when uApp is on the same device (local subscriptions) or remote device (remote subscriptions).
 
 ==== Within a uDevice
 
 .Local Subscription Flow
-image::local_subscribe.svg[Local Subscription Flow]
+[mermaid]
+ifdef::env-github[[source,mermaid]]
+----
+sequenceDiagram
+    box White Device1
+        actor uApp
+        participant uSubscription
+        participant NotificationTopic
```

Comment:
What is Notificationtopic? It is confusing to show it as another lifeline as it is not another uE. Usually I would use different arrow types to indicate if it is a request, response, publish, or notification message and I recommend you do the same in all your diagrams.